qiime2/sdk/results.py

```diff
@@ -110,3 +110,10 @@ def __repr__(self):
 
     def _asdict(self):
         return dict(zip(self._fields, self))
+
+    def result(self):
```

Comment:
Looking at this again, it actually prevents someone from making an output named "result", as they would want to be able to say:

```
results = do_action()
results.result  # suddenly a function instead of an artifact
```

The workaround is to make it a `_result`. This is the same kind of concession that `namedtuple` makes for it's methods, so while it's strange, it's basically the same contract as the other instance methods.